libs/elasticsearch/langchain_elasticsearch/_sync/retrievers.py

```diff
@@ -13,34 +13,374 @@
+
+        llm = ChatOllama(model="llama3", temperature=0)
+
+        # Create a chain that retrieves documents and then generates a response
+        def format_docs(docs):
+            # Format documents for the prompt
+            return "\n\n".join(doc.page_content for doc in docs)
+
+        system_prompt = (
+            "You are an assistant for question-answering tasks. "
+            "Use the following pieces of retrieved context to answer "
+            "the question. If you don't know the answer, say that you "
+            "don't know. Use three sentences maximum and keep the "
+            "answer concise."
+            "\n\n"
+            "Context: {context}"
+        )
+
+        prompt = ChatPromptTemplate.from_messages([
+            ("system", system_prompt),
+            ("human", "{question}"),
+        ])
+
+        chain = (
+            {"context": retriever | format_docs, "question": RunnablePassthrough()}
+            | prompt
+            | llm
+        )
+
+        result = chain.invoke("what is the answer to this question?")
+        ```
+
+    For synchronous applications, use the `ElasticsearchRetriever` class.
+    For asynchronous applications, use the `AsyncElasticsearchRetriever` class.
     """
 
-    es_client: Elasticsearch
+    client: Elasticsearch
     index_name: Union[str, Sequence[str]]
     body_func: Callable[[str], Dict]
     content_field: Optional[Union[str, Mapping[str, str]]] = None
     document_mapper: Optional[Callable[[Mapping], Document]] = None
 
-    def __init__(self, **kwargs: Any) -> None:
-        super().__init__(**kwargs)
+    def __init__(
+        self,
+        index_name: Union[str, Sequence[str]],
+        body_func: Callable[[str], Dict],
+        *,
+        content_field: Optional[Union[str, Mapping[str, str]]] = None,
+        document_mapper: Optional[Callable[[Mapping], Document]] = None,
+        client: Optional[Elasticsearch] = None,
+        es_url: Optional[str] = None,
+        es_cloud_id: Optional[str] = None,
+        es_user: Optional[str] = None,
+        es_api_key: Optional[str] = None,
+        es_password: Optional[str] = None,
+    ) -> None:
+        # Create client from credentials if needed (BEFORE super().__init__)
+        if client is not None:
+            es_connection = client
+        elif es_url is not None or es_cloud_id is not None:
+            es_connection = create_elasticsearch_client(
+                url=es_url,
+                cloud_id=es_cloud_id,
+                api_key=es_api_key,
+                username=es_user,
+                password=es_password,
+            )
+        else:
+            raise ValueError(
+                "Either 'client' or credentials (es_url, es_cloud_id, etc.) "
+                "must be provided."
+            )
+
+        # Apply user agent
+        es_connection = with_user_agent_header(es_connection, "langchain-py-r")
+
+        # Pass ALL Pydantic fields to super().__init__() so Pydantic sets them
```

Comment:
Why a reference to Pydantic here? I don't understand the significance of it. The parent class is a langchain class I think. I wasn't even aware there is Pydantic somewhere in the mix.